maintenance/populateLogUsertext.php

```diff
@@ -53,16 +53,34 @@ protected function doDBUpdates() {
 		$end += $this->mBatchSize - 1;
 		$blockStart = $start;
 		$blockEnd = $start + $this->mBatchSize - 1;
+
 		while ( $blockEnd <= $end ) {
 			$this->output( "...doing log_id from $blockStart to $blockEnd\n" );
-			$cond = "log_id BETWEEN $blockStart AND $blockEnd AND log_user = user_id";
-			$res = $db->select( array( 'logging', 'user' ),
-				array( 'log_id', 'user_name' ), $cond, __METHOD__ );
+			$res = $db->select(
+				'logging',
+				[ 'log_id', 'log_user' ],
+				"log_id BETWEEN $blockStart AND $blockEnd",
+				__METHOD__
+			);
+
+			$ids = [];
+
+			foreach ( $res as $row ) {
+				$ids[] = $row->log_user;
+			}
+
+			$res->rewind();
+
+			$users = User::whoAre( $ids );
```

Comment:
Will `$ids` have unique values or should we first call `array_unique` over them?

Reply:
`User::whoAre()` uses MySQL's `DISTINCT`.  We can consider using `array_unique` inside the method.  We are discussing that in the relevant pull request.